### ISSUE TITLE
Shutdown camera in VideoCapture java wrapper is not possible

### ISSUE THREAD

**Issue by abratchik** [opencv/opencv#7418]
##### System information (version)

- OpenCV => 3.1
- Operating System / Platform => Mac OSX
- Compiler => XCode Version 8.0 (8A218a)
##### Detailed description

I tried to experiment with VideoCapture class in java. It looks like there is a bug - it is possible to open the camera but closing is not possible. 
##### Steps to reproduce

This code leads to java crash ():

```
    VideoCapture cap = new VideoCapture();

    cap.open(0); // camera is opening up 

    Mat img = new Mat();

    cap.read(img);

    cap.release(); // java crashes

```

**Comment by AlexanderWolz** [comment]
I have the same problem as well but I could Identify that it has to be something inside the Java wrappers on macOS.

I compiled the latest opencv-master to my MacBook (macOS Sierra 10.12 x86_64) and to my Raspberry Pi 3+ (Linux ARM) using 
```
cmake -D CMAKE_BUILD_TYPE=RELEASE -D CMAKE_INSTALL_PREFIX=/usr/local -D BUILD_SHARED_LIBS=OFF -D BUILD_PYTHON_SUPPORT=ON ..
```
Then I tried abratchik's code snippet -> Running the same code did work for the Linux arch but crashes on OSX.

So I think that this issue is probably related to Apple's change from QTKit to AVFoundation. (Could be related to issue #7266) 

The underlying C-Implementation seems to work though, as I could successfully release the VideoCapture using Python (at least it does not throw an exception)

```.python
import cv2
cap = cv2.VideoCapture(0)
ret, frame = cap.read()
cap.release()
print("finished")
```
